frontend/rpm/template.go

```diff
@@ -293,45 +294,78 @@ func (w *specWrapper) BuildSteps() fmt.Stringer {
 
 func (w *specWrapper) PreUn() fmt.Stringer {
 	b := &strings.Builder{}
-	b.WriteString("%preun\n")
 
-	if w.Spec.Artifacts.Systemd != nil {
-		keys := dalec.SortMapKeys(w.Spec.Artifacts.Systemd.Units)
-		for _, servicePath := range keys {
-			serviceName := filepath.Base(servicePath)
-			fmt.Fprintf(b, "%%systemd_preun %s\n", serviceName)
-		}
+	if w.Spec.Artifacts.Systemd == nil {
+		return b
+	}
+
+	if len(w.Spec.Artifacts.Systemd.Units) == 0 {
+		return b
+	}
+
+	if strings.HasSuffix(w.Target, containerSuffix) {
```

Comment:
I thought we'd decided we weren't going to have different behavior based on the target?

Reply:
I forgot to push my revert commit! Should be fixed now